packages/components/src/navigation/test/index.js

```diff
@@ -45,92 +51,291 @@ const testNavigation = ( { activeItem, rootTitle, showBadge } = {} ) => (
+		expect( screen.queryByRole( 'heading' ) ).not.toBeInTheDocument();
 
-		rerender( testNavigation( { rootTitle: 'Home' } ) );
+		rerender( <TestNavigation rootTitle="Home" /> );
 
-		const rootTitle = screen.getByRole( 'heading' );
-		expect( rootTitle.textContent ).toBe( 'Home' );
+		expect( screen.getByRole( 'heading' ) ).toBeInTheDocument();
+		expect( screen.getByRole( 'heading' ) ).toHaveTextContent( 'Home' );
 	} );
 
-	it( 'should render badges', async () => {
-		render( testNavigation( { showBadge: true } ) );
+	it( 'should render badges', () => {
+		render( <TestNavigation showBadge /> );
 
 		const menuItem = screen.getAllByRole( 'listitem' );
 		expect( menuItem[ 0 ].textContent ).toBe( 'Item 1' + '21' );
 	} );
 
-	it( 'should render menu titles when items exist', async () => {
+	it( 'should render menu titles when items exist', () => {
 		const { rerender } = render( <Navigation></Navigation> );
 
-		const emptyMenu = screen.queryByText( 'Menu title' );
-		expect( emptyMenu ).toBeNull();
+		expect( screen.queryByText( 'Menu title' ) ).not.toBeInTheDocument();
 
-		rerender( testNavigation( { rootTitle: 'Menu title' } ) );
+		rerender( <TestNavigation rootTitle="Menu title" /> );
 
-		const menuTitle = screen.queryByText( 'Menu title' );
-		expect( menuTitle ).not.toBeNull();
+		expect( screen.getByText( 'Menu title' ) ).toBeInTheDocument();
 	} );
 
 	it( 'should navigate up a level when clicking the back button', async () => {
-		render( testNavigation( { rootTitle: 'Home' } ) );
-
-		fireEvent.click( screen.getByRole( 'button', { name: 'Category' } ) );
-		let menuTitle = screen.getByRole( 'heading' );
-		expect( menuTitle.textContent ).toBe( 'Category' );
-		fireEvent.click( screen.getByRole( 'button', { name: 'Home' } ) );
-		menuTitle = screen.getByRole( 'heading' );
-		expect( menuTitle.textContent ).toBe( 'Home' );
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+
+		render( <TestNavigation rootTitle="Home" /> );
+
+		await user.click( screen.getByRole( 'button', { name: 'Category' } ) );
+
+		expect( screen.getByRole( 'heading' ) ).toHaveTextContent( 'Category' );
+
+		await user.click( screen.getByRole( 'button', { name: 'Home' } ) );
+
+		expect( screen.getByRole( 'heading' ) ).toHaveTextContent( 'Home' );
+	} );
+
+	it( 'should navigate correctly when controlled', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+
+		render( <TestNavigationControlled /> );
+
+		// check root menu is shown and item 1 is selected
+		expect(
+			screen.getByRole( 'heading', { name: 'Home' } )
+		).toBeInTheDocument();
+		expect(
+			screen.getByRole( 'button', { name: 'Item 1' } ).parentElement
+		).toHaveClass( 'is-active' );
+
+		// click Item 2, check it's selected
+		await user.click( screen.getByRole( 'button', { name: 'Item 2' } ) );
+		expect(
+			screen.getByRole( 'button', { name: 'Item 2' } ).parentElement
+		).toHaveClass( 'is-active' );
```

Comment:
Would it be worthwhile to accompany this with a check that Item 1 no longer has the `is-active` class? I notice we do something similar up in the first test of the suite, essentially confirming that there's only one active item at a time.

If yes, there may be additional places to apply this to `is-active` tests elsewhere in the test case as well.

Reply:
Along those lines, I wonder if we should instead be checking for [`aria-current`](https://testing-library.com/docs/queries/byrole#current) rather than the `.is-active` CSS. It would also make it easier to assert that no other links/buttons have an `aria-current`.

```js
expect( screen.getByRole( 'link', { current: 'page' } ).toHaveTextContent( 'Item 2' );
```

...aaaand I was confused a bit why this doesn't work, and apparently our RTL is [quite outdated](https://github.com/WordPress/gutenberg/blame/65bf5ed0d2f328fff42a024104ac59a8e44d48ab/package.json#L115)! The `current` filter [landed](https://github.com/testing-library/dom-testing-library/releases/tag/v8.2.0) in `@testing-library/dom` v8.2.0.

```
lena@Sieglinde gutenberg % npm ls @testing-library/dom
gutenberg@13.4.0 /Users/lena/Documents/Work/Automattic/gutenberg
`-- @testing-library/react@11.2.2
  `-- @testing-library/dom@7.29.0
```

And unfortunately for us, the DOM Testing Library version bump [landed](https://github.com/testing-library/react-testing-library/pull/1031) in RTL v13, which drops support for React 17! So uh... never mind I guess 🙃 

---

Side note: I bet we'll be experiencing more of these "wait why isn't this working?" moments, since the RTL docs aren't versioned. Maybe wp-components should have versioned docs at some point, too!

Reply:
Update: #41710 got merged. I rebased this PR to include those changes, and then pushed a [commit](https://github.com/WordPress/gutenberg/pull/41668/commits/ad82facc0ac00b9ca5c3a1817b350031c7d19d6a) that switches to the approach suggested above (checking for `aria-current`).

Reply:
Wow that was painless, didn't realize it was backported! Thank you for the quick version bump. Code looks super slick now 😎